### PR TITLE
feat(articles/assistant): display user messages

### PR DESCRIPTION
**Pull request by vojtaholik**
image

gif

### CODE REVIEW

File: apps/course-builder-web/src/app/articles/_components/article-assistant.tsx [hunk 1]
```diff
@@ -40,6 +42,29 @@ export function ArticleAssistant({
...
+          content: `
+          current title: ${article.title}
+          current body: ${article.body}
+          `,
+        },
+        {
+          role: ChatCompletionRequestMessageRoleEnum.User,
+          content: event.currentTarget.value,
+        },
+      ],
```
Comment (vojtaholik) [inline apps/course-builder-web/src/app/articles/_components/article-assistant.tsx:63]: Separating the current metadata from the user prompt simplifies future consideration. No parsing required, etc. Not 100% sure if the system role is correct here.

File: apps/course-builder-web/src/app/articles/_components/article-assistant.tsx [hunk 2]
```diff
@@ -128,13 +126,37 @@ function ArticleChatResponse({ requestId }: { requestId: string }) {
...
-        if (
-          messageData.body !== STREAM_COMPLETE &&
-          messageData.name == 'ai.message' &&
-          requestId === messageData.requestId
-        ) {
-          setMessages((messages) => [...messages, { body: messageData.body, requestId: messageData.requestId }])
+        if (messageData.userId) {
+          // It's a user message, add it directly.
+          setMessages((prev) => [
+            ...prev,
```
Comment (vojtaholik) [inline apps/course-builder-web/src/app/articles/_components/article-assistant.tsx:132]: this function separates individual chat messages with streaming in mind.

File: apps/course-builder-web/src/trpc/api/routers/users.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { db } from '@/server/db'
+import { createTRPCRouter, protectedProcedure, publicProcedure } from '@/trpc/api/trpc'
+import { z } from 'zod'
+
+export const usersRouter = createTRPCRouter({
+  get: publicProcedure
```
Comment (vojtaholik) [inline apps/course-builder-web/src/trpc/api/routers/users.ts:6]: simple user getter

File: apps/course-builder-web/src/app/articles/_components/article-assistant.tsx [hunk 4]
```diff
@@ -49,26 +74,12 @@ export function ArticleAssistant({
           <Textarea
             ref={textareaRef}
             className="w-full rounded-none border-0 border-b px-5 py-4 pr-10"
-            placeholder="Type a message..."
+            placeholder="Ask Assistant..."
```
Comment (joelhooks) [inline apps/course-builder-web/src/app/articles/_components/article-assistant.tsx:77]:
```suggestion
            placeholder="Direct Assistant..."
```

File: apps/course-builder-web/src/trpc/api/routers/users.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    .input(
+      z.object({
+        userId: z.string(),
+      }),
+    )
+    .query(async ({ ctx, input }) => {
+      return await db.query.users.findFirst({
```
Comment (joelhooks) [inline apps/course-builder-web/src/trpc/api/routers/users.ts:13]: this should probably be protected?

Comment (joelhooks) [reply apps/course-builder-web/src/trpc/api/routers/users.ts:13]: doesn't really matter now, but allows for public access via api

**Review by joelhooks — APPROVED** [review]

**Pull request merged**